modules/auxiliary/gather/redis_extractor.rb

```diff
@@ -0,0 +1,186 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Redis
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Redis Extractor',
+      'Description'    => %q{
+        This module connects to a Redis instance and retrieves keys and data stored.
+      },
+      'Author'     => ['Geoff Rainville noncenz[at]ultibits.com'],
+      'License'    => MSF_LICENSE,
+      'References' => [['URL', 'https://redis.io/topics/protocol']]
+
+    ))
+  end
+
+  MIN_REDIS_VERSION = '2.8.0'
+  @kv
+
+  # Recurse to assemble the full list of keys
+  def scan(offset)
+    response = redis_command('scan',offset).split("\n")
```

Comment:
Redis keys can contain newlines, so splitting based on `\n` could result in corrupted extraction:

```
>>> r.flushall()
True
>>> r.set('AAA\nBBB','CCC')
True
>>> r.set('DDD','EEE')
True
>>> r.set('FFF','GGG')
True
```

Then in MSF:

```
 Key  Value
 ---  -----
 $3   unknown key type +none
 AAA  unknown key type +none
 DDD  EEE
```